pkg/controllers/awsloadbalancercontroller/credentials_request.go

```diff
@@ -0,0 +1,184 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	cco "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+const (
+	credentialRequestName      = "aws-load-balancer-credentials-request"
+	credentialRequestNamespace = "openshift-cloud-credential-operator"
+)
+
+// currentCredentialsRequest returns true if credentials request exists.
+func (r *AWSLoadBalancerControllerReconciler) currentCredentialsRequest(ctx context.Context, name types.NamespacedName) (bool, *cco.CredentialsRequest, error) {
+	cr := &cco.CredentialsRequest{}
+	if err := r.Client.Get(ctx, name, cr); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, err
+	}
+	return true, cr, nil
+}
+
+func (r *AWSLoadBalancerControllerReconciler) ensureCredentialsRequest(ctx context.Context, namespace string, _ *albo.AWSLoadBalancerController) error {
+
+	credReq := createCredentialsRequestName(credentialRequestName)
+
+	reqLogger := log.FromContext(ctx).WithValues("credentialsrequest", credReq)
+	reqLogger.Info("reconciling credentials secret for aws-load-balancer instance")
+
+	exists, current, err := r.currentCredentialsRequest(ctx, credReq)
+	if err != nil {
+		reqLogger.Error(err, "failed to find existing credential request")
+		return err
+	}
+
+	// The secret created will be in the operator namespace.
+	secretRef := createCredentialsSecretRef(namespace)
+
+	desired, err := desiredCredentialsRequest(ctx, credReq, secretRef)
+	if err != nil {
+		reqLogger.Error(err, "failed to build desired credential request")
+		return err
+	}
+
+	if !exists {
+		if err := r.createCredentialsRequest(ctx, desired); err != nil {
+			return err
+		}
+		_, _, err = r.currentCredentialsRequest(ctx, credReq)
+		return err
+	}
+
+	if updated, err := r.updateCredentialsRequest(ctx, current, desired); err != nil {
+		reqLogger.Error(err, "failed to update credential request")
+		return err
+	} else if updated {
+		_, _, err = r.currentCredentialsRequest(ctx, credReq)
+		return err
+	}
+
+	return nil
+}
+
+func (r *AWSLoadBalancerControllerReconciler) createCredentialsRequest(ctx context.Context, desired *cco.CredentialsRequest) error {
+	if err := r.Client.Create(ctx, desired); err != nil {
+		return fmt.Errorf("failed to create aws-load-balancer credentials request %s: %w", desired.Name, err)
+	}
+	return nil
+}
+
+func (r *AWSLoadBalancerControllerReconciler) updateCredentialsRequest(ctx context.Context, current *cco.CredentialsRequest, desired *cco.CredentialsRequest) (bool, error) {
```

Comment:
Would be nice to have a comment to be able to easily find out the meaning of the returned `bool`.

Reply:
fixed